### PR TITLE
[otlib] Add a config option to write GPOS 7 lookups

### PR DESCRIPTION
**Pull request by khaledhosny**
See the inline comment and option documentation

### CODE REVIEW

File: Lib/fontTools/otlLib/builder.py [hunk 1]
```diff
@@ -369,10 +369,19 @@ def build(self):
...
         # correctly process GPOS7 lookups, so for now we force contextual
         # positioning lookups to be chaining (GPOS8).
-        if self.subtable_type == "Pos":  # horrible separation of concerns breach
+        #
+        # This seems to be fixed as of macOS 13.2, but we keep disabling this
+        # for now until we are no longer concerned about old macOS versions.
+        # But we allow people to opt-out of this with the config key below.
+        write_gpos7 = self.font.cfg.get("fontTools.otlLib.builder:WRITE_GPOS7")
+        # horrible separation of concerns breach
+        if not write_gpos7 and self.subtable_type == "Pos":
```
Comment (anthrotype) [inline Lib/fontTools/otlLib/builder.py:384]: What exactly is "horrible"?

**Review by anthrotype — COMMENTED** [review]
LGTM

File: Lib/fontTools/config/__init__.py [hunk 2]
```diff
@@ -57,3 +57,18 @@ class Config(AbstractConfig):
     parse=Option.parse_optional_bool,
     validate=Option.validate_optional_bool,
 )
+
+Config.register_option(
+    name="fontTools.otlLib.builder:WRITE_GPOS7",
+    help=dedent(
+        """\
+        macOS before 13.2 didn’t support GPOS LookupType 7 (none-chaining
```
Comment (dscorbett) [inline Lib/fontTools/config/__init__.py:65]: “none-chaining” should be “non-chaining”.

File: Lib/fontTools/config/__init__.py [hunk 3]
```diff
@@ -57,3 +57,18 @@ class Config(AbstractConfig):
...
+        ContextPos lookups), so FontTools.otLib.builder disables a file size
```
Comment (dscorbett) [inline Lib/fontTools/config/__init__.py:66]: “FontTools.otLib.builder” should be “fontTools.otlLib.builder”.

File: Lib/fontTools/otlLib/builder.py (hunk 1 above)
Comment (khaledhosny) [reply Lib/fontTools/otlLib/builder.py:384]: That is an existing comment, I moved it up to improve code formatting.

Comment (anthrotype) [reply Lib/fontTools/otlLib/builder.py:384]: I know it preexisted, but I don't understand why it is horrible breach of separation of concerns

Comment (khaledhosny) [reply Lib/fontTools/otlLib/builder.py:384]: That is a question to @simoncozens I guess.

**Comment by anthrotype** [comment]
can we merge this? I'd like to cut a fonttools release soon as I'd like to use some newly added functionality in ufo2ft, and this PR seems ready to me

**Pull request merged**